uniffi_meta/src/lib.rs

```diff
@@ -182,10 +182,72 @@ impl ConstructorMetadata {
     }
 }
 
+// A "sub-type" which represents all the "self" receivers we support.
+// It would ideally be `Type`
```

Comment:
I agree, making `Method::receiver` be `Type` seems nicer to me, why can't we do that?  The only downside that I can see is that the code would need to handle `receiver` being an invalid type, but worst-case we can just panic.

Reply:
There are 2 reasons this is tricky:
* We can't reasonably support an "unnamed" type. I almost thought about `Type` having a sub-enum to handle named types, but
* The other problem is that for `Object` we don't know the `ObjectImpl`, and I'm not sure how we could get it.